spec/features/application/when_user_is_admin_navbar_shows_spec.rb

```diff
@@ -0,0 +1,30 @@
+require "rails_helper"
+
+feature "when user is admin navbar shows" do
```

Comment:
This might be more appropriate as a view spec. What do you think?

Reply:
@elle I was going down the controller good path bad path, which by the sound of it doesn't apply to view specs.
Issues with view specs:
One issue is trying to implement current_user in view specs - it isn't being recognised.
If I use allow(view).to receive(:current_user).and_return(user) or return(true) I get the error: 
`@rendered_views={}>> does not implement: current_user` preceded by all the view rendering.
Is there a way to implement current_user in application controller to log the current_user in?
The other issue is if I don't put anything in to log the user in I get past the above issue and get the error:
`undefined local variable or method `current_user'`, so once I get to the view it doesn't recognise current_user? I'm hoping that if I can find a way to login the user this will be taken care of?

Reply:
Looking at the RSpec docco, we should be able to stub our views that way:
https://relishapp.com/rspec/rspec-rails/docs/view-specs/view-spec#view-specs-can-stub-a-helper-method

But for some reason it wasn't working for me with the partial. Even so, I've added the view specs in `b79d92a`. Have a look and let me know what you think

Reply:
Wow @elle, mind officially blown, thank you so much for putting those together for me, I'm not sure how long it would have taken googling to find the answer, as I wasn't going down the right path! I will take a better look when I get home!